src/Guzzle6HttpAdapter.php

```diff
@@ -45,24 +45,40 @@ public function sendRequest(RequestInterface $request)
     {
         try {
             return $this->client->send($request);
-        } catch (RequestException $e) {
-            throw $this->createException($e);
+        } catch (Guzzle\SeekException $e) {
+            throw new RequestException($e->getMessage(), $request, $e);
+        } catch (Guzzle\TransferException $e) {
+            throw $this->handleTransferException($e);
         }
     }
 
     /**
      * Converts a Guzzle exception into an Httplug exception.
      *
-     * @param RequestException $exception
+     * @param Guzzle\TransferException $exception
      *
-     * @return Exception
+     * @return RequestException
      */
-    private function createException(RequestException $exception)
+    private function handleTransferException(Guzzle\TransferException $exception)
     {
-        if ($exception->hasResponse()) {
-            return new HttpException($exception->getMessage(), $exception->getRequest(), $exception->getResponse(), $exception);
+        if ($exception instanceof Guzzle\ConnectException || $exception instanceof Guzzle\TooManyRedirectsException) {
+            return new NetworkException($exception->getMessage(), $exception->getRequest(), $exception);
+        }
+
+        if ($exception instanceof Guzzle\RequestException) {
```

Comment:
i would not do an elseif. we have a return above, the complexity to read the code is simpler this way.